src/API.jl

```diff
@@ -1275,6 +1275,29 @@ function precompile(ctx::Context, pkgs::Vector{PackageSpec}; internal_call::Bool
         end
     end
 
+    stderr_buffers = Dict{Base.PkgId,IOBuffer}()
```

Comment:
IOBuffer is actually not a valid type to use for output (or often with Tasks in general) since it is lossy, I just keep forgetting to implement the error for it to instruct people toward the correct one. The correct type is Base.BufferStream and should mostly be a drop-in replacement (since they are both buffered IO objects).

Reply:
Ok, but I really don't understand the nuance between the two, and the IOBuffer docstring examples seem to be quite similar to how I use it https://docs.julialang.org/en/v1/base/io-network/#Base.IOBuffer

But, I've switched it out here.

Reply:
> should mostly be a drop-in replacement

Unfortunately I hit the gap
```
MethodError: no method matching take!(::Base.BufferStream)
```

Reply:
Yeah, `take!` is not something defined for most `IO`

Reply:
Normally the method to call is `readline(io)` or `read(io, String)`

Reply:
Ok. `read` on an empty BufferStream blocks. What's the equivalent to `take!` which doesn't block on an empty IOBuffer?

Reply:
`readavailable` also blocks on an empty BufferStream. But perhaps I can query `bytesavailable` first.

```
julia> bytesavailable(iob)
0

julia> println(iob, "hello")

julia> bytesavailable(iob)
6
```

Reply:
updated

Reply:
readline would be the normal function to use. If you are worried about blocking, you might be doing the work in the wrong task

Reply:
Ok, I think I have it in better shape now. thanks